editing/edit-with-branch-versioning/src/main/java/com/esri/samples/edit_with_branch_versioning/EditWithBranchVersioningController.java

```diff
@@ -0,0 +1,352 @@
+        }
+        createVersionButton.setText("Create version");
+        createVersionButton.setDisable(false);
+      }
+    });
+  }
+
+  /**
+   * Apply local edits to the service geodatabase and switch branch version
+   */
+  @FXML
+  private void handleSwitchVersionButtonClicked() {
+    if (serviceGeodatabase.getVersionName().equals(userCreatedVersion)) {
+      // if the user created version has local edits, apply the edits to the service geodatabase
+      if (serviceGeodatabase.hasLocalEdits()) {
+        ListenableFuture<List<FeatureTableEditResult>> resultOfApplyEdits = serviceGeodatabase.applyEditsAsync();
+        resultOfApplyEdits.addDoneListener(() -> {
+          try {
+            // check if the server edit was successful
+            List<FeatureTableEditResult> edits = resultOfApplyEdits.get();
+            if (edits == null || edits.size() == 0) {
+              new Alert(Alert.AlertType.ERROR, "Error applying edits on server").show();
+            } else {
+              // if the edits were successful, switch to the default version
+              switchVersion(defaultVersion);
+            }
+          } catch (InterruptedException | ExecutionException e) {
+            new Alert(Alert.AlertType.ERROR, "Error applying edits on server").show();
+          }
+        });
+      } else {
+        // if there are no local edits, switch to the default version
+        switchVersion(defaultVersion);
+      }
+    } else if (serviceGeodatabase.getVersionName().equals(defaultVersion)) {
+      // if the current version is the default version, switch to the user created version
+      switchVersion(userCreatedVersion);
+    }
+  }
+
+  /**
+   * Switch the active branch version
+   *
+   * @param versionName name of the version to switch to
+   */
+  private void switchVersion(String versionName) {
+    ListenableFuture<Void> switchVersionResult = serviceGeodatabase.switchVersionAsync(versionName);
+    switchVersionResult.addDoneListener(() -> {
+      // check if the active version has switched successfully and update the UI
+      if (serviceGeodatabase.getVersionName().equals(versionName)) {
+        currentVersionLabel.setText("Current version: " + serviceGeodatabase.getVersionName());
+        editFeatureVBox.setDisable(true);
+      } else {
+        new Alert(Alert.AlertType.ERROR, "Error switching version.").show();
+      }
+    });
+  }
+
+  /**
+   * Select a feature if one exists where the user clicked
+   *
+   * @param point location where the user clicked
+   */
+  private void selectFeature(Point2D point){
+    ListenableFuture<IdentifyLayerResult> results = mapView.identifyLayerAsync(featureLayer, point, 1, false);
+    results.addDoneListener(() -> {
+      try {
+        IdentifyLayerResult layer = results.get();
+        List<GeoElement> identified = layer.getElements();
+        if (!identified.isEmpty()) {
+          GeoElement element = identified.get(0);
```

Comment:
We should give the variables in here more recognisable names so it's easier to tell what class the objects are:
```suggestion
    ListenableFuture<IdentifyLayerResult> identifyLayerResultFuture = mapView.identifyLayerAsync(featureLayer, point, 1, false);
    identifyLayerResultFuture.addDoneListener(() -> {
      try {
        IdentifyLayerResult identifyLayerResult = identifyLayerResultFuture.get();
        List<GeoElement> identifiedElements = identifyLayerResult.getElements();
        if (!identifiedElements.isEmpty()) {
          GeoElement element = identifiedElements.get(0);
```